internal/attestation/certcache/cached_client.go

```diff
@@ -44,25 +44,32 @@ func (c *CachedHTTPSGetter) Get(url string) ([]byte, error) {
 	default:
 	}
 
-	// Don't cache CRLs. Unlike VCEKs, these can change over time and the KDS
-	// doesn't rate-limit requests to these.
-	canCache := !crlURL.MatchString(url)
-
-	if canCache {
-		if cached, ok := c.cache.Get(url); ok {
-			c.logger.Debug("Get cached", "url", url)
-			return cached, nil
+	if crlURL.MatchString(url) {
+		// For CRLs always query. When request failure, fallback to cache.
+		c.logger.Debug("Request CRL", "url", url)
+		res, err := c.HTTPSGetter.Get(url)
+		if err != nil {
+			if cached, ok := c.cache.Get(url); ok {
+				c.logger.Debug("CRL request failed, fallback to cached CRL", "url", url)
+				return cached, nil
+			}
+			c.logger.Debug("CRL request failed and CRL was not found in cache", "url", url)
+			return nil, err
 		}
+		c.cache.Set(url, res)
+		return res, nil
```

Comment:
I think we can have less branches if we return early. Also I think we want a Warn or Info if the server could not be reached. 
For example:
```suggestion
		res, err := c.HTTPSGetter.Get(url)
		if err == nil {
			c.cache.Set(url, res)
			return res, nil
		}
		c.logger.Warn("could not reach server", "error", err)
		// Could not reach server, check cache
		if cached, ok := c.cache.Get(url); ok {
			c.logger.Debug("CRL request failed, fallback to cached CRL", "url", url)
			return cached, nil
		}
		c.logger.Debug("CRL request failed and CRL was not found in cache", "url", url)
		return nil, err
		
		
```